Lib/test/test_fcntl.py

```diff
@@ -152,8 +152,8 @@ def test_flock(self):
 
         self.assertRaises(ValueError, fcntl.flock, -1, fcntl.LOCK_SH)
         self.assertRaises(TypeError, fcntl.flock, 'spam', fcntl.LOCK_SH)
-
-    @unittest.skipIf(platform.system() == "AIX", "AIX returns PermissionError")
+    @unittest.skipIf(platform.system() == "AIX" or platform.system() == "OS400",
+                     "AIX and OS400 returns PermissionError")
```

Comment:
I think that this would not need to be skipped at all if we applied the same fix to lockf as we do for flock.

Reply:
Fixes in 052edb5